lib/puppet/gettext/config.rb

```diff
@@ -30,12 +47,43 @@ def self.translation_mode(conf_path)
     end
   end
 
+  # Prevent future gettext initializations
+  def self.disable_gettext
+    @gettext_disabled = true
+  end
+
   # Attempt to initialize the gettext-setup gem
   # @param path [String] to gettext config file
   # @param file_format [Symbol] translation file format to use, either :po or :mo
   # @return true if initialization succeeded, false otherwise
   def self.initialize(conf_file_location, file_format)
-    # Bypass gettext until we can resolve a performance regression related to it, PUP-8009.
-    return false
+    return if @gettext_disabled
+
+    unless file_format == :po || file_format == :mo
+      raise Puppet::Error, "Unsupported translation file format #{file_format}; please use :po or :mo"
+    end
+
+    return false unless @gettext_found
+
+    if conf_file_location && File.exists?(conf_file_location)
```

Comment:
Prefer to use `Puppet::FileSystem.exist?` - https://github.com/puppetlabs/puppet/blob/master/lib/puppet/file_system.rb#L170-L172